### PR TITLE
[JENKINS-31425] should be able to specify multiple users or groups for the submitter of an input step

### PR DESCRIPTION
**Pull request by escoem**
submitter now allows multiple users/groups separated by ','

@reviewbybees please

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-submitter.html [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <div>
-    User ID or <em>external</em> group name of person or people permitted to respond to the input.
+    User IDs and/or <em>external</em> group names of person or people permitted to respond to the input, splitted by ','.
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-submitter.html]: “Splitted” is not a word in English. “Split” is the passive participle, but probably you meant “separated” in this case anyway.

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java [hunk 2]
```diff
@@ -145,6 +145,29 @@ public void test_cancel_run_by_input() throws Exception {
...
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy().
+                // Only give "alice" basic privs. That's normally not enough to Job.CANCEL, only for the fact that "alice"
+                // is listed as the submitter.
+                        grant(Jenkins.READ, Job.READ).everywhere().to("alice").
+                // Only give "bob" basic privs. That's normally not enough to Job.CANCEL, only for the fact that "bob"
+                // is listed as the submitter.
+                        grant(Jenkins.READ, Job.READ).everywhere().to("bob").
+                // Give "charlie" basic privs.  That's normally not enough to Job.CANCEL, and isn't listed as submiter.
+                        grant(Jenkins.READ, Job.READ).everywhere().to("charlie"));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java]: `to` can take a list.

File: src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-submitter.html (hunk 1 above)
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-submitter.html]: Best to clarify that whitespace is not trimmed: `alice, bob` will not match `alice`.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java [hunk 3]
```diff
@@ -116,10 +119,13 @@ public boolean canSubmit() {
      */
     @Deprecated
     public boolean canSettle(Authentication a) {
-        if (submitter==null || a.getName().equals(submitter))
+        if (submitter==null)
+            return true;
+        final Set<String> submitters = Sets.newHashSet(submitter.split(","));
+        if (submitters.contains(a.getName()))
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java]: I think being able to add a space after the coma is a usual thing for a user to do to make the code more readable.
Whilst some realms do allow groups with a trailing space - its probably not normal - so could we trim the submitter before checking so we can use
`"bob, jane, freddy`"?

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java]: Well, the behavior is at least documented, so I think this is good enough.

**Comment by vrenjith** [comment]
Is there a date when this will be released?

**Comment by rompic** [comment]
really lookig forward for the next release!

**Comment by jglick** [comment]
On my to-do list; would like to check JENKINS-38380 first.

**Comment by rompic** [comment]
Thanks for reloading It. An important step forward!

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java (hunk 3 above)
Comment (batmat) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java]: Improvement proposed as https://github.com/jenkinsci/pipeline-input-step-plugin/pull/30 @jtnord BTW

**Pull request merged**